composer/algorithms/low_precision_groupnorm/low_precision_groupnorm.py

```diff
@@ -0,0 +1,99 @@
+
+from composer.algorithms.warnings import NoEffectWarning
+from composer.core import Algorithm, Event, Precision, State
+from composer.loggers import Logger
+from composer.utils import module_surgery
+
+log = logging.getLogger(__name__)
+
+
+def _cast_if_autocast_enabled(hidden_states):
+    if not torch.is_autocast_enabled():
+        return hidden_states
+    else:
+        return torch.cuda.amp.autocast_mode._cast(hidden_states, torch.get_autocast_gpu_dtype())
+
+
+class LPGroupNorm(torch.nn.GroupNorm):
+
+    def __init__(self, layer):
+        super().__init__(
+            num_groups=layer.num_groups,
+            num_channels=layer.num_channels,
+            eps=layer.eps,
+            affine=layer.affine,
+        )
+
+        with torch.no_grad():
+            self.weight.copy_(layer.weight)
+            self.bias.copy_(layer.bias)
+
+    def forward(self, x):
+        module_device = x.device
+        downcast_x = _cast_if_autocast_enabled(x)
+        downcast_weight = _cast_if_autocast_enabled(self.weight)
+        downcast_bias = _cast_if_autocast_enabled(self.bias)
+        with torch.autocast(enabled=False, device_type=module_device.type):
+            return F.group_norm(downcast_x, self.num_groups, downcast_weight, downcast_bias, self.eps)
+
+
+def to_LPGroupNorm(layer: torch.nn.Module, module_index: int) -> LPGroupNorm:
+    assert isinstance(layer,
+                      torch.nn.GroupNorm), 'The replacement policy will look for all instances of torch.nn.GroupNorm'
+    return LPGroupNorm(layer)
+
+
+def apply_low_precision_groupnorm(model, optimizers: Union[torch.optim.Optimizer, Sequence[torch.optim.Optimizer]],
+                                  precision: Precision):
+
+    if (precision != Precision.AMP_FP16 and precision != Precision.AMP_BF16):
+        warnings.warn(NoEffectWarning('Low Precision GroupNorm only applies to AMP_FP16 and AMP_BF16 precisions.'))
+        return model
+
+    policy: Dict[Type[torch.nn.Module], module_surgery.ReplacementFunction] = {torch.nn.GroupNorm: to_LPGroupNorm}
+
+    replaced_instances = module_surgery.replace_module_classes(module=model, optimizers=optimizers, policies=policy)
+    print(f'Replaced {len(replaced_instances)} instances of GroupNorm with LowPrecisionGroupNorm')
+    if len(replaced_instances) == 0:
+        warnings.warn(NoEffectWarning('No instances of torch.nn.GroupNorm found.'))
+    log.info(f'Successfully replaced {len(replaced_instances)} instances of GroupNorm with LowPrecisionGroupNorm')
+
+
+class LowPrecisionGroupNorm(Algorithm):
+    """
+    Replaces all instances of `torch.nn.GroupNorm` with `composer.algorithms.low_precision_groupnorm.low_precision_groupnorm.LPGroupNorm`.
+
+    LPGroupNorm is a thin wrapper around `torch.nn.GroupNorm` which forces the layer to run in lower precision (torch.float16 or torch.bfloat16)
+    if autocast is enabled. This algorithm has no effect in FP32 or DeepSpeed FP16 mode, where autocast is disabled.
```

Comment:
```suggestion
    Replaces all instances of :class:`torch.nn.GroupNorm` with :class:`.LPGroupNorm`.

    `LPGroupNorm` is a thin wrapper around :class:`torch.nn.GroupNorm`_ which forces the layer to run in 
    lower precision (torch.float16 or torch.bfloat16) if autocast is enabled. This algorithm has no effect in 
    FP32 or DeepSpeed FP16 mode, where autocast is disabled.
```
(lines under 120 chars, internal links)